hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/SaslRpcServer.java

```diff
@@ -225,18 +226,22 @@ public enum AuthMethod {
     SIMPLE((byte) 80, ""),
     KERBEROS((byte) 81, "GSSAPI"),
     @Deprecated
-    DIGEST((byte) 82, SaslMechanismFactory.getMechanism()),
-    TOKEN((byte) 82, SaslMechanismFactory.getMechanism()),
+    DIGEST((byte) 82, SaslMechanismFactory::getMechanism),
+    TOKEN((byte) 82, SaslMechanismFactory::getMechanism),
     PLAIN((byte) 83, "PLAIN");
 
     /** The code for this method. */
     public final byte code;
-    public final String mechanismName;
+    private final Supplier<String> mechanismName;
 
     private AuthMethod(byte code, String mechanismName) { 
+      this(code, () -> mechanismName);
+    }
+
+    AuthMethod(byte code, Supplier<String> mechanismName) {
       this.code = code;
       this.mechanismName = mechanismName;
-      LOG.info("{} {}: code={}, mechanism=\"{}\"",
+      LOG.debug("{} {}: code={}, mechanism=\"{}\"",
```

Comment:
AFAICT this ultimately gets called at roughly the same point during client initialization as previously, it just doesn't show up because the debug level has been reduced to DEBUG from INFO.

Reply:
So apart from the log level the only difference is that this will pick up changes in the config
if a new SaslRPCServer created at a later time ?

Reply:
if you aren't using SASL stuff `saslMechanismFactory::getMechanism` doesn't get invoked.

Reply:
> ... this will pick up changes in the config if a new SaslRPCServer is created at a later time ?

It will load the config only when `SaslMechanismFactory.getMechanism` is called.  So, it won't be loaded with UGI.